src/modules/RoutingModule.cpp

```diff
@@ -20,6 +20,12 @@ bool RoutingModule::handleReceivedProtobuf(const meshtastic_MeshPacket &mp, mesh
         if ((nodeDB->getMeshNode(mp.from) == NULL || !nodeDB->getMeshNode(mp.from)->has_user) &&
             (nodeDB->getMeshNode(mp.to) == NULL || !nodeDB->getMeshNode(mp.to)->has_user))
             return false;
+    } else if (owner.is_licensed && nodeDB->getMeshNode(mp.from) != NULL && nodeDB->getMeshNode(mp.from)->has_user &&
```

Comment:
Minor thing, but it seems rather inefficient to call `getMeshNode()` three times for the same node, especially with large NodeDBs. Probably better to define a `fromNode`. Same optimization can be done in `perhapsDecode()`.

Reply:
Good point. I'll make a specific method to check I think.